Slave4

53rus opened the pull request.

53rus committed:
Работа на проверку

53rus committed:
3.3 Работа на проверку

53rus committed:
3.4 Работа на проверку

53rus committed:
3.5 Работа на проверку

53rus committed:
3.6 Работа на проверку

53rus committed:
4.1 Работа на проверку

53rus committed:
4.2 Работа на проверку

53rus committed:
4.2 Работа на проверку (исправление)

53rus committed:
4.2 Работа на проверку (исправление - 2)

53rus committed:
4.3 Работа на проверку

isavinov commented on src/main/resources/liquibase/scripts/course-four-lesson-three.sql:
```diff
@@ -0,0 +1,7 @@
+-- changeset miliuschenko:1
+CREATE INDEX student_name_index ON student (name);
+
+-- changeset miliuschenko:2
```
Разные changeset лучше делать в разных файлах

53rus committed:
4.4 Работа на проверку, не дает собрать Jar проект, почему то перестали проходить тесты

53rus committed:
4.5 Работа на проверку.
-Добавить эндпоинт для получения всех имен всех студентов, чье имя начинается с буквы А.
-Создать эндпоинт, который будет возвращать средний возраст всех студентов
-Создать эндпоинт, который будет возвращать самое длинное название факультета.
-Создан эндпоинт с модифицированной логикой

pavelilyukevichsky commented on src/main/java/ru_hogwarts_school/service/StudentService.java:
```diff
@@ -121,4 +123,22 @@ public Collection<Student> getLastFiveStudent() {
+    public Collection<String> getAllStudentByNameStartedA() {
+        logger.info("Was invoked method for get all student whose name begins A");
+
+        return studentRepository.findAll().stream().filter(student -> student.getName().startsWith("А"))
```
не правильно оформлен стрим. вот так надо
```
studentRepository.findAll().stream()
                              .filter(student -> student.getName().startsWith("А"))
                              .map(student -> student.getName().toUpperCase())
                              .sorted()
                              .collect(Collectors.toList());
```

53rus committed:
4.6 Работа на проверку.
- Создан эндпоинт GET /students/print-parallel. Эндпоинт GET /students/print-parallel обращается к сервису, где первые 2 студента распечатываются в Main потоке, а 4 следующих имени распечатываются в двух других потоках
 - Создан эндпоинт GET /students/print-synchronized. Создан синхронизированный метод для вывода имен студентов в консоль. Эндпоинт GET /students/print-synchronized обращается к сервису, где первые 2 студента распечатываются в Main потоке, а 4 следующих имени распечатываются через синхронизированный метод.
